staging/src/k8s.io/dynamic-resource-allocation/cel/compile.go

```diff
@@ -253,10 +276,11 @@ func mustBuildEnv() *environment.EnvSet {
 		}
 		return result
 	}
+
 	deviceType := apiservercel.NewObjectType("kubernetes.DRADevice", fields(
-		field(driverVar, apiservercel.StringType, true),
-		field(attributesVar, apiservercel.NewMapType(apiservercel.StringType, apiservercel.NewMapType(apiservercel.StringType, apiservercel.AnyType, resourceapi.ResourceSliceMaxAttributesAndCapacitiesPerDevice), resourceapi.ResourceSliceMaxAttributesAndCapacitiesPerDevice), true),
-		field(capacityVar, apiservercel.NewMapType(apiservercel.StringType, apiservercel.NewMapType(apiservercel.StringType, apiservercel.QuantityDeclType, resourceapi.ResourceSliceMaxAttributesAndCapacitiesPerDevice), resourceapi.ResourceSliceMaxAttributesAndCapacitiesPerDevice), true),
+		field(driverVar, driverType, true),
```

Comment:
Just brainstorming on how to ensure someone doesn't accidentally forget to set a MaxElements on a type..  

Option: `field()` takes the MaxElements as an argument, you could define consts for the values provided.
Option: `field()` takes a type alias to `apiservercel.AnyType` and this package provides a function that requires the MaxElements argument and returns the alias type.
Option: `field()` asserts that passed types (at least the string and bytes typed ones, maybe others) have MaxElements set.

Reply:
This isn't limited to fields, nor to this particular source code.

Someone else in some entirely different package might do `apiservercel.NewMapType(apiservercel.StringType, apiservercel.StringType, myMaxNumberOfKeys)` without realizing that they should use some custom string type with `MaxElements` set.

To disarm this footgun everywhere, those pre-defined types would have to be removed in favor of constructor methods which force passing a `maxElements` parameters.

Alternatively, the size estimator could refuse to use an unset `MaxElements`. But that then only fails if someone is careful with unit testing.

Reply:
True.. I'd be OK with opening a separate issue for the pre-defined types -> constructor method idea.

WDYT @cici37 is this something the CEL contributors would be willing to work on?